### ISSUE TITLE
Reuse Matan's Medium.com lazy-loading article

### ISSUE THREAD

**Issue by filiph** [dart-lang/site-webdev#582]
Matan's [article about lazy loading in AngularDart](https://medium.com/@matanlurey/lazy-loading-with-angular-dart-14f58004f988) is something that should be part of the documentation. 

Contrast and compare with angular.io documentation, which makes a big deal about lazy loading. Our documentation doesn't ever mention that lazy-loading is part of the language itself.

In general: it's awesome that Matan and others can write and publish these with zero barrier. In parallel, we should monitor and appropriate some of that content. Let's start [here](https://medium.com/@matanlurey).

**Comment by kwalrath** [comment]
@matanlurey does v3 have additional deferred support?

**Comment by matanlurey** [comment]
Not officially (`@experimental`). We can have more details out soon.

**Comment by filiph** [comment]
Let's focus on what we have right now, unless `!deferred` is _just_ around the corner. I think the language support is pretty impressive, and at least comparable (hopefully much better) than what the TypeScript world has.

**Comment by kwalrath** [comment]
Goodness, Matan, I didn't realize you'd published [so many articles](https://medium.com/@matanlurey)!

**Comment by kwalrath** [comment]
angular.io covers/mentions lazy loading in a few places:
* [deployment guide](https://angular.io/docs/ts/latest/guide/deployment.html)
* [router guide](https://angular.io/docs/ts/latest/guide/router.html)
* [style guide](https://angular.io/docs/ts/latest/guide/style-guide.html)
* places that don't apply to Dart, such as the NgModules [guide](https://angular.io/docs/ts/latest/guide/ngmodule.html) and [FAQ](https://angular.io/docs/ts/latest/cookbook/ngmodule-faq.html), and assorted API reference docs

@chalin we should talk about how to cover this.

**Comment by chalin** [comment]
Yes. Let's also discuss relative priorities of [Q2 work items](527).

> language support is pretty impressive

But, if I understand correctly, this is a Dart language feature; and Matan's article illustrates how it can be used in the context of an AngularDart app. Is that correct?  Can we have a heads up of about which other deferred-related features are in the pipeline (in a private communication if necessary). This will help us better plan the doc additions.

**Comment by chalin** [comment]
Folding #436 into here.

**Comment by zanuka** [comment]
@chalin Is someone currently working on documentation for lazy / async loading on this page? [Asynchronous Routing](https://webdev.dartlang.org/angular/guide/router/6)

 I've managed to get `as deferred` working in a project I'm working on, using the updated docs in the [component-loading faq](https://github.com/dart-lang/angular/blob/master/doc/faq/component-loading.md) but it would be great to update the guide section for other users, since it's an extremely useful pattern to leverage for scalable/performant apps. 

I'd be happy to take on the issue of updating those docs, just let me know. Cheers.
